### ISSUE TITLE
Load fails w/ FATA[0000] Error: Untar fork/exec /usr/bin/docker: argument list too long

### ISSUE THREAD

**Issue by jwilder** [docker/docker#10426]
```
$ cat /tmp/image.tar | docker load
FATA[0000] Error: Untar fork/exec /usr/bin/docker: argument list too long 
```

You can get this error if you have a several hundred images on your host.  I had ~450 images (as reported by `docker images | wc -l`) and when loading a saved image it would fail consistently w/ this error.  

This looks like it might be related to https://github.com/docker/docker/blob/092d52281d20e3456677ac11d781abf1cd0bdd8f/graph/load.go#L41 which constructs a list of all image IDs on the host which is then serialized to JSON and passed as an argument to a subsequent exec call.

After deleting my images down to ~250, `docker load` started working again.

```
$ docker version
Client version: 1.4.1
Client API version: 1.16
Go version (client): go1.3.3
Git commit (client): 5bc2ff8
OS/Arch (client): linux/amd64
Server version: 1.4.1
Server API version: 1.16
Go version (server): go1.3.3
Git commit (server): 5bc2ff8
```
```
$ docker info
Containers: 194
Images: 1526
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Dirs: 1921
Execution Driver: native-0.2
Kernel Version: 3.13.0-24-generic
Operating System: Ubuntu 14.04 LTS
CPUs: 8
Total Memory: 15.36 GiB
Name: laptop
ID: PIYC:2PIA:5YSP:XY2N:YE2Z:KO5T:YCEJ:VWQJ:MUUO:KC54:GKES:CKFT
Debug mode (server): true
Debug mode (client): false
Fds: 10
Goroutines: 10
EventsListeners: 0
Init Path: /usr/bin/docker
Docker Root Dir: /var/lib/docker
Registry: [https://index.docker.io/v1/]
```

```
$ uname -a
Linux laptop 3.13.0-24-generic #47-Ubuntu SMP Fri May 2 23:30:00 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by crosbymichael** [comment]
I marked this as a bug.  The fix should be straight forward for anyone wanting to work on this.

**Comment by anandkumarpatel** [comment]
+1 just saw this as well
```
root@charizard:/home/ubuntu# docker images | wc -l
971
```
```
root@charizard:/home/ubuntu# docker -D version
Client version: 1.4.1
Client API version: 1.16
Go version (client): go1.3.3
Git commit (client): 5bc2ff8
OS/Arch (client): linux/amd64
Server version: 1.4.1
Server API version: 1.16
Go version (server): go1.3.3
Git commit (server): 5bc2ff8

root@charizard:/home/ubuntu# docker -D info
Containers: 1740
Images: 3356
Storage Driver: aufs
 Root Dir: /docker/aufs
 Dirs: 6840
Execution Driver: native-0.2
Kernel Version: 3.13.0-29-generic
Operating System: Ubuntu 14.04 LTS
CPUs: 32
Total Memory: 58.81 GiB
Name: charizard
ID: HDXE:A3BL:XLN7:SFJY:AANA:PFAD:B7QC:YNF3:XSD4:D23X:CKS3:IF3K
Debug mode (server): true
Debug mode (client): true
Fds: 660
Goroutines: 1182
EventsListeners: 1
Init Path: /usr/bin/docker
Docker Root Dir: /docker
WARNING: No swap limit support

root@charizard:/home/ubuntu# uname -a
Linux charizard 3.13.0-29-generic #53-Ubuntu SMP Wed Jun 4 21:00:20 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by resouer** [comment]
Much simpler, I think use a loop instead of passing all the images as arguments to shell also fix this problem.

**Comment by fishtek** [comment]
Hello guys, I've been running into this issue (initially when using the [docker-squash](https://github.com/jwilder/docker-squash) tool by @jwilder , however also with simply doing a docker save, docker load.  

[I noticed that the fix was merged into the docker 1.6.0 release candidate branch](https://github.com/docker/docker/pull/10439), so [I used the 1.6.0-rc5 binary](https://test.docker.com/builds/Linux/x86_64/docker-1.6.0-rc5) and tried this out again, however I'm still running into the issue.  Is there a chance the fix regressed again in later RC's of the 1.6.0 branch?  Or perhaps I'm running into a different issue.

Here is what I've done along w/ the standard docker info, let me know if I can test anything or if there is something more you need:

```bash
sysadmin@slave2:~$ sudo docker save 7ffeadff89aa > test.image.tar
sysadmin@slave2:~$ ls -lh
total 249M
-rw-rw-r-- 1 sysadmin sysadmin 222M Apr 10 12:31 test.image.tar
sysadmin@slave2:~$ sudo docker rmi 7ffeadff89aa
Deleted: 7ffeadff89aad854b5db2b8c96f4ca26a376408683b391136754fd15974e8635
Deleted: 8e076f1b2d249f7b601e3f683ba2d158899d77b8fcd464e89a0849f76683ac7b
Deleted: cf48f0e0977afe9e748ba73a03999a536a563b46a1a6f47876935d548cd4eef2
Deleted: 6ef9af2152278464499813b2776343d471212655497f6d02b727085a6d979220
Deleted: f8a3943e4dfc813ea4bf9caaca4361569957fb4083be099c8c4cc49e0b948c77
Deleted: 38a0d72b0a889d22ad6c7262986270766dfafb1c4aa13390990ef4fcff15aad7
Deleted: dcb456c824cb953502f54ad5be3aef6313cc584a847a1bce589e9e755bbb7851
Deleted: 74f544f5ff4de93696b1b6b1942bfb94b5605c43f51f492c3deb524bb7d9642d
sysadmin@slave2:~$ sudo docker images | wc -l
372
sysadmin@slave2:~$ sudo docker load -i test.image.tar 
FATA[0000] Error response from daemon: Untar fork/exec /usr/bin/docker: argument list too long 
sysadmin@slave2:~$ sudo docker info && sudo docker version
Containers: 1
Images: 1994
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 1996
 Dirperm1 Supported: false
Execution Driver: native-0.2
Kernel Version: 3.13.0-49-generic
Operating System: Ubuntu precise (12.04.5 LTS)
CPUs: 4
Total Memory: 3.86 GiB
Name: slave2
ID: L2Y4:4XL5:J56E:SWOQ:EV7G:FBTU:4ACZ:4LBB:RAQW:FBNP:MT32:7JH7
WARNING: No swap limit support
Client version: 1.6.0-rc5
Client API version: 1.18
Go version (client): go1.4.2
Git commit (client): fc4825d
OS/Arch (client): linux/amd64
Server version: 1.6.0-rc5
Server API version: 1.18
Go version (server): go1.4.2
Git commit (server): fc4825d
OS/Arch (server): linux/amd64

```

**Comment by LK4D4** [comment]
ping @estesp

**Comment by estesp** [comment]
@fishtek sorry about poor assumptions on the original fix--the correct fix is in PR #12276 just submitted

**Comment by estesp** [comment]
@fishtek @jwilder this is now fixed in upstream master, but did not make the 1.6.0 release; so unless there is a 1.6.1, it will be in 1.7.0; apologies for not catching the problem with the initial fix.
